cli/install_ee/install_ee.go

```diff
@@ -0,0 +1,163 @@
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/util"
+	"github.com/tarantool/tt/cli/version"
+	"golang.org/x/term"
+)
+
+const eeSource string = "https://download.tarantool.io/enterprise/"
+
+type creds struct {
+	username string
+	password string
+}
+
+func getCredsInteractive() (creds, error) {
+	res := creds{}
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Printf("Enter username: ")
+	resp, err := reader.ReadString('\n')
+	if err != nil {
+		return res, err
+	}
+	res.username = strings.TrimSpace(resp)
+
+	fmt.Printf("Enter password: ")
+	bytePass, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return res, err
+	}
+	res.password = strings.TrimSpace(string(bytePass))
+
+	return res, nil
+}
+
+func getCredsFromFile(path string) (creds, error) {
+	res := creds{}
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return res, err
+	}
+
+	re := regexp.MustCompile("(?P<user>.*):(?P<pass>.*)")
+	matches := util.FindNamedMatches(re, strings.TrimSpace(string(data)))
+
+	if len(matches) == 0 {
+		return res, fmt.Errorf("Corrupted credentials.")
+	}
+
+	res.username = matches["user"]
+	res.password = matches["pass"]
+
+	return res, nil
+}
+
+func getCreds(cliOpts *config.CliOpts) (creds, error) {
+	if cliOpts.EE == nil || (cliOpts.EE != nil && cliOpts.EE.CredPath == "") {
+		return getCredsInteractive()
+	}
+
+	return getCredsFromFile(cliOpts.EE.CredPath)
+}
+
+func getTarballName(data string) (string, error) {
+	re := regexp.MustCompile(">(.*)<")
+
+	parsedData := re.FindStringSubmatch(data)
+	if len(parsedData) == 0 {
+		return "", fmt.Errorf("Cannot parse tarball name")
+	}
+
+	return parsedData[1], nil
+}
+
+// parseIndex collects a list of versions for the host architecture.
+func parseIndex(data *[]byte) ([]version.Version, error) {
+	versions := []version.Version{}
+	matchRe := ""
+
+	arch, err := util.GetArch()
+	if err != nil {
+		return nil, err
+	}
+
+	// Bundles without specifying the architecture are all x86_64.
+	if arch == "x86_64" {
+		matchRe = ".*>tarantool-enterprise-bundle-" +
+			"(.*-g[a-f0-9]+-r[0-9]{3})(?:-linux-x86_64)?\\.tar\\.gz<.*"
+	} else {
+		matchRe = ".*>tarantool-enterprise-bundle-" +
+			"(.*-g[a-f0-9]+-r[0-9]{3})(?:-linux-" + arch + ")\\.tar\\.gz<.*"
+	}
+
+	re := regexp.MustCompile(matchRe)
+	parsedData := re.FindAllStringSubmatch(strings.TrimSpace(string(*data)), -1)
+	if len(parsedData) == 0 {
+		return nil, fmt.Errorf("Cannot parse versions")
+	}
+
+	for _, entry := range parsedData {
+		version, err := version.GetVersionDetails(entry[1])
+		if err != nil {
+			return nil, err
+		}
+		version.Str = entry[1]
+		version.Tarball, err = getTarballName(entry[0])
+		if err != nil {
+			return nil, err
+		}
+		versions = append(versions, version)
+	}
+
+	version.SortVersions(versions)
+
+	return versions, nil
+}
+
+func FetchVersions(cliOpts *config.CliOpts) ([]version.Version, error) {
```

Comment:
add a description pls, also many functions in this file also doesn't have one. I'm generally okay with that, but don't know how other members will think about it.

Reply:
Done